### DISCUSSION TITLE
Could this resolve CSS variables?

### DISCUSSION THREAD

**Discussion by splitbrain** [MyIntervals/PHP-CSS-Parser#747] [General]
I have a style sheet that relies on CSS variables (aka custom properties). Now I want to generate PDFs with mpdf, which doesn't understand those. Instead of writing a custom style sheet for the PDF export I was wondering if there's a way to automate.

Would it be possible to use this library to create stylesheet that resolves all the variables to their appropriate values? Has anyone done this before and could give some pointers or example code?

**Comment by JakeQZ** [comment]
You can probably use our sister package [Emogrifier](https://github.com/MyIntervals/emogrifier) for this.  This inlines the CSS as style attributes directly in the HTML, resolving all CSS rules.  Originally intended for email clients that have poor CSS support, it should work just as well to help out other applications with limited CSS support.

However, support for CSS custom properties has only just been added, and is not yet in a released version.  You could try with the `-dev` package to see if it works for you.  A new release is imminent, to support PHP 8.4, and will also include CSS variables support.

The readme on the GitHub page includes the details for resolving CSS variables.

Let us know if you have any questions or issues.

**Reply by splitbrain** [reply]
That looks excellent! Thanks for the pointer.

**Reply by JakeQZ** [reply]
> A new release is imminent, to support PHP 8.4, and will also include CSS variables support.

This is now released: https://github.com/MyIntervals/emogrifier/releases/tag/v7.3.0

**Comment by oliverklee** [comment]
Closing, as we now have both an answer as well as an Emogrifier release.
